src/iam/iam-base.ts

```diff
@@ -673,4 +677,31 @@ export class IAMBase {
       })
     });
   }
+
+  private async initContracts() {
+    const { chainId } = await this._provider.getNetwork();
+    const {
+      ensRegistryAddress, ensResolverAddress, didContractAddress, cacheServerUrl
+    } = IAMBase.chainConfigs[chainId];
+    this._registrySetting = {
+      address: didContractAddress,
+      abi: abi1056,
+      method: Methods.Erc1056
+    };
+    this._ensRegistryAddress = ensRegistryAddress;
+    this._ensResolverAddress = ensResolverAddress;
+    this._ensRegistry = EnsRegistryFactory.connect(ensRegistryAddress, this._provider);
+    this._ensResolver = PublicResolverFactory.connect(ensResolverAddress, this._provider);
+    cacheServerUrl && (this._cacheClient = new CacheServerClient({ url: cacheServerUrl }));
+  }
+
+  /**
+   * @description - should be invoked before iniitialization
+   * 
+   * @param chainId 
+   * @param config 
+   */
+  static setChainConfig(chainId: number, config: ChainConfig) {
```

Comment:
Not a big deal, but I think it would be nice if could selectively override a property for a given config, instead of needing to specific the whole config.  Maybe consider accepting a `Partial<ChainConfig>`.

Reply:
Good idea